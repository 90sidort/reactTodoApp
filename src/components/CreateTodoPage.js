import React from "react";
import TodoForm from "./TodoForm";
import { connect } from "react-redux";
import { startAddTodo } from "../actions/todos";

export class CreateTodoPage extends React.Component {
  onSubmit = (todo) => {
    this.props.startAddTodo(todo);
    this.props.history.push("/");
  };
  render() {
    return (
      <div>
        <div className="pageHeader">
          <div className="summaryContent">
            <h1 className="pageHeaderTitle">Create Todo</h1>
          </div>
        </div>
        <div className="content-container">
          <TodoForm onSubmit={this.onSubmit} />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddTodo: (todo) => dispatch(startAddTodo(todo)),
});

export default connect(undefined, mapDispatchToProps)(CreateTodoPage);
