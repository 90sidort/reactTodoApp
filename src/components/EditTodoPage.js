import React from 'react';
import { connect } from 'react-redux'
import TodoForm from './TodoForm'
import { editTodo, removeTodo } from '../actions/todos'

export class EditTodoPage extends React.Component {
    onSubmit = (todo) => {
        this.props.editTodo(this.props.todo.id, todo)
        this.props.history.push('/')
    }
    onRemove = () => {
        this.props.removeTodo({id: this.props.todo.id})
        this.props.history.push('/')
    }
    render() {
        return (
            <div>
                <TodoForm 
                    todo={this.props.todo}
                    onSubmit={this.onSubmit}
                />
                <button
                    onClick={this.onRemove}
                >
                    Remove
                </button>
            </div>
        )
    }
}

const mapStateToProps = (state, props) => {
    return {
        todo: state.todos.find((todo) => todo.id === props.match.params.id)
    }
}

const mapDispatchToProps = (dispatch) => ({
    editTodo: (id, todo) => dispatch(editTodo(id, todo)),
    removeTodo: (data) => dispatch(removeTodo(data))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditTodoPage)