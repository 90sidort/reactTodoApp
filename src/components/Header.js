import React from 'react';
import { NavLink } from 'react-router-dom'

const Header = () => (
    <header>
        <h1>Todo app</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink>
        <NavLink to="/create" activeClassName="is-active" exact={true}>Create todo</NavLink>
        <NavLink to="/help" activeClassName="is-active" exact={true}>Help</NavLink>
    </header>
)

export default Header