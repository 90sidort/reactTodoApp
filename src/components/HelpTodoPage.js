import React from 'react';

const HelpTodoPage = () => (
    <div>
        This is my todo help page.
    </div>
)

export default HelpTodoPage