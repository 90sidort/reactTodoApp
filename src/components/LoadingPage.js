import React from "react";

const LoadingPage = () => (
  <div className="loader">
    <img src="/images/loader.gif" />
    <p>Loading...</p>
  </div>
);

export default LoadingPage;
