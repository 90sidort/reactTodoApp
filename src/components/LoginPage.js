import React from "react";
import { connect } from "react-redux";
import { startLogin, startLogUser } from "../actions/auth";

export const LoginPage = ({ startLogin, startLogUser }) => {
  return (
    <div className="box-layout">
      <div className="box-layout__box">
        <h1 className="box-layout__title">Todo App</h1>
        <p>It's time to get more organized!</p>
        <button className="buttonBlue" onClick={startLogin}>
          Login with Google
        </button>
        <button className="buttonBlue" onClick={startLogUser}>
          Test user
        </button>
      </div>
    </div>
  );
};

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin()),
  startLogUser: () => dispatch(startLogUser()),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
