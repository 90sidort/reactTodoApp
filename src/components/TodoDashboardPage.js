import React from 'react';
import ConnectedTodoList from './TodoList'
import TodoListFilters from '../components/TodoListFilters'

const TodoDashboardPage = () => (
    <div>
        This is my TodoDashboardPage.
        <TodoListFilters/>
        <ConnectedTodoList/>
    </div>
)

export default TodoDashboardPage