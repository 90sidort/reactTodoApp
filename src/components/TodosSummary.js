import React from 'react'
import { connect } from 'react-redux'
import countByPriority from '../selectors/priorityCounter'
import getVisibleTodos from '../selectors/todos'

export const TodosSummary = ({visibleTodos}) => {
    const count = countByPriority(visibleTodos)
    const sum = count.reduce((sum, value) => sum + value, 0)
    const countForm = visibleTodos.length === 1 ? 'todo' : 'todos'
    return (
        <div>
            <p>Viewing {sum} {countForm}.</p>
            <p>{count[0]} low, {count[1]} regular and {count[2]} high priotity</p>
        </div>
    )
}

const mapStateToProps = (state) => {
    const filteredTodos = getVisibleTodos(state.todos, state.filters)

    return {
        visibleTodos: filteredTodos
    }
}

export default connect(mapStateToProps)(TodosSummary)