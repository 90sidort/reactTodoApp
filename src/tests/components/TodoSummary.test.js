import { TodosSummary } from '../../components/TodosSummary'
import React from 'react'
import { shallow } from 'enzyme'
import mockTodos from '../fixtures/todosFixture'

test('Should render component correctly with no todos', () => {
    const result = shallow(<TodosSummary visibleTodos={[]}/>)
    expect(result).toMatchSnapshot()
})

test('Should render component correctly with one todo', () => {
    const result = shallow(<TodosSummary visibleTodos={[mockTodos[0]]}/>)
    expect(result).toMatchSnapshot()
})

test('Should render component correctly with multiple todos', () => {
    const result = shallow(<TodosSummary visibleTodos={mockTodos}/>)
    expect(result).toMatchSnapshot()
})