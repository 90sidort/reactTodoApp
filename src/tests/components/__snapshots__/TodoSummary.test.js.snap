// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render component correctly with multiple todos 1`] = `
<div>
  <p>
    Viewing 
    3
     
    todos
    .
  </p>
  <p>
    1
     low, 
    1
     regular and 
    1
     high priotity
  </p>
</div>
`;

exports[`Should render component correctly with no todos 1`] = `
<div>
  <p>
    Viewing 
    0
     
    todos
    .
  </p>
  <p>
    0
     low, 
    0
     regular and 
    0
     high priotity
  </p>
</div>
`;

exports[`Should render component correctly with one todo 1`] = `
<div>
  <p>
    Viewing 
    1
     
    todo
    .
  </p>
  <p>
    1
     low, 
    0
     regular and 
    0
     high priotity
  </p>
</div>
`;
